crates/swc_common/src/input.rs

```diff
@@ -67,6 +67,13 @@ impl<'a> Input for StringInput<'a> {
         self.iter.clone().nth(2).map(|i| i.1)
     }
 
+    #[inline]
+    fn peek_iter(&self) -> Box<dyn Iterator<Item = char> + '_> {
```

Comment:
This allocates unconditionally

Reply:
I think we don't need this.
Adding `is_str` is enough, and adding this is a breaking change anyway

Reply:
Should I add a `is_str` method to the Input?

Reply:
Yeah, I think it would be better.